### ISSUE TITLE
`ruff server` does not load new configuration files when they are created

### ISSUE THREAD

**Issue by snowsignal** [astral-sh/ruff#11766]
If you create a new `pyproject.toml` / `ruff.toml` configuration file in your workspace, we reload any existing configuration but do not 'register' the new configuration file.

This has been a known limitation of `Index::reload_settings`:

```rust
/// Reloads relevant existing settings files based on a changed settings file path.
/// This does not currently register new settings files.
```

**Comment by dhruvmanila** [comment]
I'm not sure whether this is actually a problem or not because it's suppose to be handled on the client side. The client will send us request on file events (create, change, delete). So, this should work.

**Comment by dhruvmanila** [comment]
As seen in the test plan of https://github.com/astral-sh/ruff/pull/12253, I'm creating a new `pyproject.toml` file in a nested folder and the diagnostics are being refreshed accordingly. At the end, I delete the file and it's refreshed as per the root `pyproject.toml` file.
